sdk/spring/azure-spring-integration-core/src/main/java/com/azure/spring/integration/core/AbstractDefaultMassageHandler.java

```diff
@@ -0,0 +1,146 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.spring.integration.core;
+
+import com.azure.spring.integration.core.api.PartitionSupplier;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.expression.EvaluationContext;
+import org.springframework.expression.Expression;
+import org.springframework.expression.common.LiteralExpression;
+import org.springframework.integration.expression.ValueExpression;
+import org.springframework.integration.handler.AbstractMessageProducingHandler;
+import org.springframework.integration.support.DefaultErrorMessageStrategy;
+import org.springframework.integration.support.ErrorMessageStrategy;
+import org.springframework.messaging.Message;
+import org.springframework.messaging.MessageChannel;
+import org.springframework.util.Assert;
+import org.springframework.util.StringUtils;
+import org.springframework.util.concurrent.ListenableFutureCallback;
+
+import java.util.Optional;
+
+/**
+ * Base class of outbound adapter to publish to azure backed messaging service
+ */
+public abstract class AbstractDefaultMassageHandler extends AbstractMessageProducingHandler {
```

Comment:
After discuss with @saragluna , we do not need to create a abstract parent class for `CompletableFuture` and `reactor`. Because we will delete `CompletableFuture` after we upgrade `servicebus` to tack 2. So for now, we just keep duplicated codes.